packages/components/src/Datalist/Datalist.component.js

```diff
@@ -0,0 +1,361 @@
+	 * ENTER: select a suggestion, persist the value, or submit the form
+	 * UP/DOWN: update the active suggestion index
+	 * @param event The keydown event
+	 * @param highlightedItemIndex The previous focused suggestion index
+	 * @param newHighlightedItemIndex The new focused suggestion index
+	 * @param highlightedSectionIndex The previous focused section index
+	 * @param newHighlightedSectionIndex The new focused section index
+	 */
+	onKeyDown(event, params) {
+		const {
+			highlightedItemIndex,
+			highlightedSectionIndex,
+			newHighlightedItemIndex,
+			newHighlightedSectionIndex,
+		} = params;
+		switch (event.which) {
+			case keycode.codes.esc:
+				event.preventDefault();
+				this.resetValue();
+				break;
+			case keycode.codes.enter:
+				if (!this.state.suggestions) {
+					break;
+				}
+				event.preventDefault();
+				if (Number.isInteger(highlightedItemIndex)) {
+					// suggestions are displayed and an item has the focus : we select it
+					this.onSelect(event, {
+						itemIndex: highlightedItemIndex,
+						sectionIndex: highlightedSectionIndex,
+					});
+				} else if (this.state.value !== this.state.previousValue) {
+					// there is no focused item and the current value is not persisted
+					// we persist it
+					this.updateValue(event, this.state.value, true);
+				}
+				// resetting suggestions to close dropdown
+				this.resetSuggestions();
+				break;
+			case keycode.codes.down:
+				event.preventDefault();
+				if (!this.state.suggestions) {
+					// display all suggestions when they are not displayed
+					this.updateSuggestions();
+					this.updateSelectedIndexes(this.state.value);
+					break;
+				}
+				this.setState({
+					focusedItemIndex: newHighlightedItemIndex,
+					focusedSectionIndex: newHighlightedSectionIndex,
+				});
+				break;
+			case keycode.codes.up:
+				event.preventDefault();
+				this.setState({
+					focusedItemIndex: newHighlightedItemIndex,
+					focusedSectionIndex: newHighlightedSectionIndex,
+				});
+				break;
+			default:
+				break;
+		}
+	}
+
+	/**
+	 * Select an item in suggestions list
+	 * @param event The select event
+	 * @param itemIndex The item index in suggestions list
+	 */
+	onSelect(event, { sectionIndex, itemIndex }) {
+		let newValue = this.state.suggestions[itemIndex];
+		if (this.props.multiSection) {
+			newValue = this.state.suggestions[sectionIndex].suggestions[itemIndex];
+		}
+		this.updateValue(event, newValue, true);
+	}
+
+	/**
+	 * Reset the focused item and section
+	 */
+	resetSelection() {
+		this.setState({
+			focusedItemIndex: undefined,
+			focusedSectionIndex: undefined,
+		});
+	}
+
+	/**
+	 * Update the focused item and section given the current value
+	 * Managing the two cases (multi section and single section)
+	 */
+	updateSelectedIndexes(value) {
+		if (this.props.multiSection) {
+			const groups = this.props.titleMap;
+			for (let sectionIndex = 0; sectionIndex < groups.length; sectionIndex++) {
```

Comment:
I prefer: groups.forEach((group, index) => {});

Reply:
did that at first, but cannot break the loop that way (@jsomsanith review), so i switched back to simple for in order to be able to break it.

Reply:
I made him change to avoid to process all the items when the wanted one has been found.
I'll suppress the lint not to block this PR, it doesn't make sense, let's propose to remove it.

Reply:
so why not a find ?